tests/unit/cli/test_base.py

```diff
@@ -17,11 +17,11 @@
 from click.testing import CliRunner
 
 from zenml.cli.base import init
-from zenml.repository import LOCAL_CONFIG_DIRECTORY_NAME
+from zenml.constants import REPOSITORY_DIRECTORY_NAME
```

Comment:
I'm cognizant of the irony of me seeing this change and using it to suggest the following:

Initially the intent of this PR was to remove the "Repository" class entirely and handle all of its tasks on a profile or global config level. Although we then realized that we can't avoid keeping some configuration inside of every project's folder.... I wonder if it might not still be helpful to remove the name "Repository" at least. Really, whatever this directory still is, I don't think it's a repository. Maybe rename the Repository to LocalConfig, to contrast it with GlobalConfig? Overall, I think the naming of the classes in this PR could benefit from a bit more clarity, just being more descriptive of what they actually do. 

Currently:
* GlobalConfig
* ConfigProfile
* Repository

Suggestion
* GlobalConfiguration
* ProfileConfiguration
* LocalConfiguration? 

Maybe something like that. Unless I'm misunderstanding what these objects do

Reply:
We can't rename Repositoy for two main reasons:
1. Repository has been part of the public API and is a very central concept that is already heavily in use in code by our users everywhere. Sure, we can still have something like a `Repository = LocalConfiguration` dummy placeholder for it (but see my other reason).
2. Repository is more than just configuration. It is a facade for many different things, such as managing stacks, retrieving pipeline run results and I imagine it will extend to cover all other information involved in the Zen Service communication (Users, Teams, maybe even Secrets ?).

I do agree in principle that `Repository` is not a good name for it anymore and that `LocalConfiguration` is a set of concerns that can be detached from it and moved elsewhere. That's what inspired me to create that `BaseGlobalConfiguration` abstract class that is implemented both by GlobalConfig and Repository, but that is a more convoluted change that should be done separately, if not out of concern for the size of this PR, then at least for time considerations (this PR needs to go in the next release).